app/Exceptions/Handler.php

```diff
@@ -29,8 +30,14 @@ class Handler extends ExceptionHandler
      *
      * @throws \Throwable
      */
-    public function render($request, Throwable $exception): Response
+    public function render($request, Throwable $exception): JsonResponse
     {
-        return response($this->apiException($request, $exception));
+        try {
+            $error = $this->apiException($request, $exception);
+
+            return response()->json($error, $error->getStatusCode());
+        } catch (Throwable $e) {
+            return parent::render($request, $exception);
```

Comment:
According to the base Handler class, parent::render here will return Response instead of JsonResponse, so it goes against declared return type.